gcc/config.gcc:
  We need this alignment thing to get some defines that the code relies on
gcc/config/i386/darwin64.h:
  Don't forget about crt1.o, makes it hard to link something together

--- gcc_42-5566/gcc/config.gcc
+++ gcc_42-5566/gcc/config.gcc
@@ -1051,6 +1051,7 @@
 	# Deleted with_arch
 	# APPLE LOCAL mainline
 	with_cpu=${with_cpu:-core2}
+	extra_options="${extra_options} i386/darwin.opt"
 	tmake_file="t-darwin ${cpu_type}/t-darwin64 t-slibgcc-darwin"
 	tm_file="${tm_file} ${cpu_type}/darwin64.h"
 	;;
--- gcc_42-5566/gcc/config/i386/darwin64.h
+++ gcc_42-5566/gcc/config/i386/darwin64.h
@@ -30,6 +30,7 @@
 
 #undef SUBTARGET_EXTRA_SPECS
 #define SUBTARGET_EXTRA_SPECS                                   \
+  DARWIN_EXTRA_SPECS \
   { "darwin_arch", DARWIN_ARCH_SPEC },                          \
   { "darwin_crt2", "" },                                        \
   { "darwin_subarch", DARWIN_SUBARCH_SPEC },
